internal/worker/server.go

```diff
@@ -299,7 +299,41 @@ func (s *Server) OSBuildJobInfo(id uuid.UUID, result *OSBuildJobResult) (*JobInf
 	if jobInfo.JobType != JobTypeOSBuild {
 		return nil, fmt.Errorf("expected %q, found %q job instead", JobTypeOSBuild, jobInfo.JobType)
 	}
+	return s.prepareOSBuildJobREsult(jobInfo, result)
+}
 
+// Returns a JobInfo and fills up the result with this information:
+//
+//    - JobError
+//    - Success
+//    - OSBuildOutput.Success (if it has an OSBuildOutput)
+//    - OSBuildOutput.Error (if it has an OSBuildOutput)
+//    - TargetResults
+//
+//Additionally, OSBuildOutput stays nil if no data associated to it is in the DB
+func (s *Server) AsyncOSBuildJobInfo(id uuid.UUID, result *OSBuildJobResult) (*JobInfo, error) {
```

Comment:
I think I don't understand this method's name. Does `Async` mean asynchronous? However, I don't see anything making this method asynchronous than `OSBuildJobInfo`.